public/app/core/components/NavBar/NavBarMenu.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useRef } from 'react';
+import { GrafanaTheme2, NavModelItem } from '@grafana/data';
+import { CustomScrollbar, Icon, IconButton, IconName, useTheme2 } from '@grafana/ui';
+import { FocusScope } from '@react-aria/focus';
+import { useOverlay } from '@react-aria/overlays';
+import { css } from '@emotion/css';
+import { NavBarMenuItem } from './NavBarMenuItem';
+
+export interface Props {
+  activeItem?: NavModelItem;
+  navItems: NavModelItem[];
+  onClose: () => void;
+}
+
+export function NavBarMenu({ activeItem, navItems, onClose }: Props) {
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+  const ref = useRef(null);
+  const { overlayProps } = useOverlay(
+    {
+      isDismissable: true,
+      isOpen: true,
+      onClose,
+    },
+    ref
+  );
+
+  return (
+    <FocusScope contain restoreFocus autoFocus>
+      <div data-testid="navbarmenu" className={styles.container} ref={ref} {...overlayProps}>
+        <div className={styles.header}>
+          <Icon name="bars" size="xl" />
+          <IconButton aria-label="Close navigation menu" name="times" onClick={onClose} size="xl" variant="secondary" />
+        </div>
+        <nav className={styles.content}>
+          <CustomScrollbar>
+            <ul>
+              {navItems.map((link, index) => (
+                <div className={styles.section} key={`${link.id}-${index}`}>
+                  <NavBarMenuItem
+                    isActive={activeItem === link}
+                    isSectionHeader
+                    onClick={() => {
+                      link.onClick?.();
+                      onClose();
+                    }}
+                    target={link.target}
+                    text={link.text}
+                    url={link.url}
+                  />
+                  {link.children?.map((childLink, childIndex) => (
+                    <NavBarMenuItem
+                      key={`${childLink.id}-${childIndex}`}
+                      icon={childLink.icon as IconName}
```

Comment:
I wish there was a better way to avoid this. Unfortunately refining a string down to a valid constant is exhausting with typescript (`if (iconName === "angle-double-down" || iconName === "angle-double-right" || ...)`).

As much as I dislike them, but maybe a type guard (`function isValidIcon(icon: string): icon is IconName`) might be better than the current situation...

(not a suggestion for this PR, just musing)